Sources/MongoSwift/BSON/Document.swift

```diff
@@ -238,6 +255,35 @@ extension Document {
         try idDoc.merge(self)
         return idDoc
     }
+
+    /// Helper function for copying elements from some source document to a destination document while
+    /// excluding a non-zero number of keys
+    internal func copyElements(to otherDoc: inout Document, excluding keys: [String]) throws {
+        guard !keys.isEmpty else {
+            throw InternalError(message: "No keys to exclude, use 'bson_copy' instead")
+        }
+
+        let cStrings: [ContiguousArray<CChar>] = keys.map { $0.utf8CString }
+        var cPtrs: [UnsafePointer<CChar>] = []
+
+        for cString in cStrings {
```

Comment:
this can also be written as a map over `cStrings`

Reply:
Done.